modules/imaging_uploader/ajax/getUploadSummary.php

```diff
@@ -18,22 +18,90 @@
  * @link     https://www.github.com/Jkat/Loris-Trunk/
  */
 
-if (!\User::singleton()->hasPermission('imaging_uploader')) {
+// Base access check - user must have either of these permissions
+// more access validation after request validation
+if (!$user->hasAnyPermission(
+    [
+        'imaging_uploader_allsites',
+        'imaging_uploader_ownsites',
+    ]
+)
+) {
     http_response_code(403);
     return;
 }
+$config        = \NDB_Factory::singleton()->config();
+$advancedperms = $config->getSetting('useAdvancedPermissions');
+$user          = \NDB_Factory::singleton()->user();
+$centerString  = implode("','", $user->getCenterIDs());
+$projectString = implode("','", $user->getProjectIDs());
+$username      = $user->getUsername();
+
 if (!validRequest()) {
     http_response_code(400);
     return;
 }
 
 $uploadId = $_POST['uploadId'];
 $summary  = $_POST['summary'] === 'true';
+$DB       = \NDB_Factory::singleton()->database();
+
+// Access Control - mimic menu filter behaviour
+// MySQL order of operations dictates that ANDs get computed before ORs which
+// means this where clause can take the follwoing forms
+// 1. WHERE mu.UploadedBy='$username' OR 1=1
+//      -> returns all records
+// 2. WHERE mu.UploadedBy='$username' OR (1=1 AND s.CenterID IN ...)
+//      -> returns records for user's sites
+// 3. WHERE mu.UploadedBy='$username' OR (1=1 AND s.ProjectID IN ...)
+//      -> returns records for user's projects
+// 4. WHERE mu.UploadedBy='$username'
+//        OR (1=1 AND s.CenterID IN ... AND s.ProjectID IN ...)
+//      -> returns records for user's sites and projects
+// 5. WHERE mu.UploadedBy='$username'
+//        OR (1=1 AND s.CenterID IN ... AND s.ProjectID IN ...)
+//        OR mu.SessionID IS NULL
+//     -> returns records for user's sites and projects and null session data
+// Other combinations are possible but order of operations still applies
+$accessQuery = "SELECT * 
+     FROM mri_upload mu 
+     LEFT JOIN session s ON (s.ID = mu.SessionID)
+     ";
+$accessWhere = " WHERE (mu.UploadedBy='$username' OR 1=1 ";
+if (!$user->hasPermission('imaging_uploader_allsites')) {
+    // Create where clause for sites
+    $where = $where . " AND s.CenterID IN ('$centerString') ";
+}
+
+if ($advancedperms === 'true') {
+    // If config setting is enabled, check the user's sites and projects
+    // site/project match + user's own uploads
+    $where = $where . " AND s.ProjectID IN ('$projectString')";
```

Comment:
This will not work if you put single quotes around `$projectString`. Why not use
```
$where = $where . " AND FIND_IN_SET(s.ProjectID, :projectString)";
```
and bind `$projectString` to `projectString` when calling `pselectRow`?

(same thing for `$centerString`)

Reply:
Oh you are right! I had not noticed that `$projectString` was set to `implode("','", $user->getProjectIDs());`. I had not noticed that each project ID was surrounded with single quote so your solution will work. All good then.